selendroid-server/src/main/java/io/selendroid/server/action/ActionChain.java

```diff
@@ -18,10 +18,11 @@
   private int pauseTime = 0;
   private int index = 0;
 
-  public ActionChain(JSONObject actionChain) throws JSONException {
+  public ActionChain(JSONObject actionChain, int i) throws JSONException {
     this.inputDevice = actionChain.getString("inputDevice");
     this.actionChain = actionChain.getJSONArray("actions");
     this.context = new ActionContext();
+    this.context.setId(i);
```

Comment:
no, i don't think in multitouchscreen is a good idea, the "id" placeholder in there appears to be for the entire multi touch action chain(s).

It seems as though this ID needs to be unique per chain event, to retain the correct pointer event to fire combinations with.